lib/tests/streamlit/connections/sql_connection_test.py

```diff
@@ -210,27 +210,29 @@ def test_repr_html_with_secrets(self):
         assert "Dialect: `postgres`" in repr_
         assert "Configured from `[connections.my_sql_connection]`" in repr_
 
-    @parameterized.expand([(DatabaseError,), (InternalError,), (OperationalError,)])
     @patch("streamlit.connections.sql_connection.SQLConnection._connect", MagicMock())
     @patch("pandas.read_sql")
-    def test_retry_behavior(self, error_class, patched_read_sql):
-        patched_read_sql.side_effect = error_class("kaboom", params=None, orig=None)
+    def test_retry_behavior(self, patched_read_sql):
+        from sqlalchemy.exc import DatabaseError, InternalError, OperationalError
 
-        conn = SQLConnection("my_sql_connection")
+        for error_class in [DatabaseError, InternalError, OperationalError]:
```

Comment:
isn't @parameterized usually favorable over `for ... in` because the test does not stop when one of the items fail and its easier to distinguish which test exactly broke based on the console output?

Reply:
Yep, but in this specific case it isn't easy to do since all the classes are imported from an optional dependency (sqlalchemy), thats why I moved into a for loop.

Reply:
I think the only alternative would be some kind of string to class mapping.. but not sure if this is a lot better

Reply:
Yeah in that case this is probably better! And string mapping does not sound too appealing either, I agree 👍